nipoppy/workflows/pipeline.py

```diff
@@ -315,7 +316,7 @@ def set_up_bids_db(
                 current=pybids_ignore_patterns,
                 new=f"^(?!/{BIDS_SUBJECT_PREFIX}({participant_id}))",
             )
-        if session_id is not None:
+        if (session_id is not None) and (session_id != FAKE_SESSION):
```

Comment:
This is not going to work because right now `session_id` doesn't have the BIDS prefix but `FAKE_SESSION` does. We can either remove the prefix from `FAKE_SESSION` (as I suggested previously) or use `nipoppy.utils.check_session_id(FAKE_SESSION)` to remove the prefix